satpy/readers/satpy_cf_nc.py

```diff
@@ -0,0 +1,112 @@
+        """Initialize file handler."""
+        super().__init__(filename, filename_info, filetype_info)
+        self.engine = None
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        return self.filename_info['start_time']
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self.filename_info.get('end_time', self.start_time)
+
+    @property
+    def sensor(self):
+        """Get sensor."""
+        nc = xr.open_dataset(self.filename, engine=self.engine)
+
+        return nc.attrs['instrument'].replace('/', '-').lower()
+
+    @property
+    def sensor_names(self):
+        """Get sensor set."""
+        return {self.sensor}
+
+    def available_datasets(self, configured_datasets=None):
+        """Add information to configured datasets."""
+        # pass along existing datasets
+        for is_avail, ds_info in (configured_datasets or []):
+            yield is_avail, ds_info
+        nc = xr.open_dataset(self.filename, engine=self.engine)
+        # get dynamic variables known to this file (that we created)
+        for var_name, val in nc.data_vars.items():
+            ds_info = dict(val.attrs)
+            ds_info['file_type'] = self.filetype_info['file_type']
+            ds_info['name'] = var_name
+            try:
+                ds_info['wavelength'] = tuple(ds_info['wavelength'])
+            except KeyError:
+                pass
+            try:
+                try:
+                    ds_info['modifiers'] = tuple(ds_info['modifiers'].split(' '))
+                except AttributeError:
+                    pass
+            except KeyError:
+                pass
+            if var_name not in ['longitude', 'latitude']:
+                ds_info['coordinates'] = ['longitude', 'latitude']
+            yield True, ds_info
+        for var_name, val in nc.coords.items():
+            ds_info = dict(val.attrs)
+            ds_info['file_type'] = self.filetype_info['file_type']
+            ds_info['name'] = var_name
+            ds_info['resolution'] = 742
+            yield True, ds_info
+
+    def get_dataset(self, ds_id, ds_info):
+        """Get dataset."""
+        logger.debug("Getting data for: %s", ds_id.name)
+        nc = xr.open_dataset(self.filename, engine=self.engine,
+                             chunks={'y': CHUNK_SIZE, 'x': CHUNK_SIZE})
+        file_key = ds_info.get('file_key', ds_id.name)
+        data = nc[file_key]
+        if file_key in nc.coords.keys():
+            data = data.drop_vars(list(nc.coords.keys()))
+        try:
+            data.attrs['wavelength'] = tuple(data.attrs['wavelength'])
+        except KeyError:
+            pass
+        try:
+            # FIXME in cf writer: this is not consitent: no modifier is (), modifiers is a string
+            try:
+                ds_info['modifiers'] = tuple(ds_info['modifiers'].split(' '))
+            except AttributeError:
+                pass
+        except KeyError:
```

Comment:
`except (AttributError, KeyError):` or is this doing something special?

Reply:
the most indented try shouldn't be needed if we fix the cf writer, so it's not supposed to be there. But yeah, I wanted to distinguish the two errors for now. The KeyError is ok, the AttributeError should happen at all...